lib/queue.rb

```diff
@@ -1,31 +1,79 @@
 class Queue
-
+  
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(10)
+    @front = -1
+    @back = -1
   end
-
+  
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if (@back == ((@front - 1) % (@store.length - 1)))
+      data = @store
+      old_length = @store.length
+      new_array = Array.new(10)
+      @store = data[@front..-1] + data[0...@front]
+      @front = 0
+      @back = old_length - 1
+    end
+    
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 0
+      @store[@back] = element
+    elsif (@back == @store.length - 1 && @front != 0)
+      @back = 0
+      @store[@back] = element
+    else
+      @back = @back + 1
+      @store[@back] = element
+    end
   end
-
+  
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1
+      print "Queue is already empty."
+    end
+    
+    data = @store[@front]
+    @store[@front] = nil
+    
+    if @front == @back
+      @front = -1
+      @back = -1
+    elsif @front == (@store.length - 1)
+      @front = 0
+    else
+      @front = @front + 1
+    end
+    
+    return data
   end
-
+  
   def front
-    raise NotImplementedError, "Not yet implemented"
+    return @front
   end
-
+  
   def size
-    raise NotImplementedError, "Not yet implemented"
+    @store.length
   end
-
+  
   def empty?
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      return true
+    else
+      return false
+    end
   end
-
+  
   def to_s
-    return @store.to_s
+    values = []
+    
+    @store.each do |i|
+      if !i.nil?
+        values << i
+      end
+    end
+    
+    return values.to_s
```

Comment:
This also only works if the back hasn't wrapped around the array

Consider:
```suggestion
    values = []
    return values if @front == @back && @back == -1
    
    current = @front
    while current != @back
      values << @store[current]
      current = (current + 1) % @store.length
    end
    
    values << @store[@back]
    
    
    return values
```